src/pkcs11/framework-pkcs15.c

```diff
@@ -1071,7 +1071,7 @@ pkcs15_init_slot(struct sc_pkcs15_card *p15card, struct sc_pkcs11_slot *slot,
 			pin_info = NULL;
 		}
 		else   {
-			if (auth->label[0] && strncmp(auth->label, "PIN", 4) != 0)
+			if (auth->label[0] && strncmp(auth->label, "PIN", 4) != 0 && strncmp(auth->label, "User Pin", 9) != 0)
```

Comment:
what do you need this change for?

Reply:
The term *User PIN* is normally used to differentiate from *SO PIN* or *Signature PIN*.

Reply:
Without this string compare, the label is copied into the tokeninfo label, which causes the inserted card to be displayed as "User Pin()" instead of "". At the same time, the closing bracket is often cut off. With the additional clause, the string is not copied and hence not shown.

Reply:
It's intended to be shown as *PIN LABEL (CARD LABEL)*. It seems your card doesn't have set `p15card->tokeninfo->label`. A better approach would be to only append *(CARD LABEL)* if `p15card->tokeninfo->label` is not empty.

Reply:
Both auth->label and p15card->tokeninfo->label are not empty, Github just removed the strings surrounded by angle brackets from my example. What the output shows is "User Pin(Example card" -where the closing bracket is oftentimes missing- instead of the desired "Example card".

Reply:
As explained above, I don't think this should be done for all cards. Either do this specifically for your card in something like `pkcs15-starcos.c` or create a configuration option that allows setting the pkcs#11 slot labels.

Reply:
I am not as well-versed in the code structure and order of execution of certain functions: Is either one of the `pkcs15-starcos.c::starcos_init_card( )` or `pkcs15-starcos.c::starcos_finalize_card( )` functions definitely going to be called After the above function `pkcs15_init_slot( )` ? In that case, one could change the card label in `pkcs15-starcos.c`, otherwise it might be overwritten again.

Reply:
please remove this change. We'll think about configuring the slot naming seperately.